position/liquidity_amounts.gno

```diff
@@ -1,49 +1,53 @@
 package position
 
-func liquidityAmountsGetLiquidityForAmount0(
-	sqrtRatioAX96 bigint,
-	sqrtRatioBX96 bigint,
-	amount0 bigint,
-) bigint {
-	if sqrtRatioAX96 > sqrtRatioBX96 {
-		sqrtRatioAX96, sqrtRatioBX96 = sqrtRatioBX96, sqrtRatioAX96
+import (
+	"gno.land/r/demo/consts"
+)
+
+// toAscendingOrder checkes if the first value is greater than
+// the second then swaps two values.
+func toAscendingOrder(a, b bigint) (bigint, bigint) {
+	if a > b {
+		return b, a
 	}
 
-	intermediate := (sqrtRatioAX96 * sqrtRatioBX96) / Q96
+	return a, b
+}
 
-	return amount0 * intermediate / (sqrtRatioBX96 - sqrtRatioAX96)
+// calcIntermediateValue computes the intermediate value
+// used in liquidity calculations.
+func calcIntermediateValue(sqrtRatioA, sqrtRatioB bigint) bigint {
+	return (sqrtRatioA * sqrtRatioB) / consts.Q96
 }
 
-func liquidityAmountsGetLiquidityForAmount1(
-	sqrtRatioAX96 bigint,
-	sqrtRatioBX96 bigint,
-	amount1 bigint,
-) bigint {
-	if sqrtRatioAX96 > sqrtRatioBX96 {
-		sqrtRatioAX96, sqrtRatioBX96 = sqrtRatioBX96, sqrtRatioAX96
-	}
+// GetLiquidityForAmount0 calculates liquidity for a given amount of token 0.
+func computeLiquidityForAmount0(sqrtRatioA, sqrtRatioB, amount0 bigint) bigint {
+	sqrtRatioA, sqrtRatioB = toAscendingOrder(sqrtRatioA, sqrtRatioB)
+	intermediate := calcIntermediateValue(sqrtRatioA, sqrtRatioB)
+	diff := sqrtRatioB - sqrtRatioA
 
-	return amount1 * Q96 / (sqrtRatioBX96 - sqrtRatioAX96)
+	// we don't need to care about division by zero here.
+	return amount0 * intermediate / diff
 }
 
-func liquidityAmountsGetLiquidityForAmounts(
-	sqrtRatioX96 bigint,
-	sqrtRatioAX96 bigint,
-	sqrtRatioBX96 bigint,
-	amount0 bigint,
-	amount1 bigint,
-) bigint {
-	var liquidity bigint
+// computeLiquidityForAmount1 calculates liquidity for a given amount of token 1.
+func computeLiquidityForAmount1(sqrtRatioA, sqrtRatioB, amount1 bigint) bigint {
+	sqrtRatioA, sqrtRatioB = toAscendingOrder(sqrtRatioA, sqrtRatioB)
```

Comment:
I think the suffix `X96` is meaningful because it tells us the bigint is actually a 96-bit fixed point integers, but I agree that they should not appear in the variable names. A better way would be defining a new X96 wrapper type so we can enforce the decimal computation at the type level. Once we adapt u256 types we can also work on it.